types/react-intl/index.d.ts

```diff
@@ -188,6 +188,7 @@ declare namespace ReactIntl {
             locale?: string;
             formats?: any;
             messages?: any;
+            textComponent?: any;
```

Comment:
I would suggest then to write type based on your experience so it will be stricter. If at some point it appeared to be wrong then it will super easy to loose it to `any`.